### DISCUSSION TITLE
Rank not updating

### DISCUSSION THREAD

**Discussion by Rizwans-github** [anuraghazra/github-readme-stats#3619] [Q&A] [answered]
### Describe the bug

I have been observing the contribution stats card for a couple of days now but it is not updating on my README.MD file although it is updated on the link itself, everything else excluding the rank is updating.

### Expected behavior

The expected behavior for the card is that its rank should change with the updates and all and when you check the link itself it shows the correct rank but not in my readme file I thought it would update after a couple of days but that's not the case here.The expected

### Screenshots / Live demo link

![image](https://github.com/anuraghazra/github-readme-stats/assets/141806496/4b5254f8-532c-43f9-823b-d850bb72ea45)
[link where it is updated](https://github-readme-stats.vercel.app/api?username=rizwans-github&rank_icon=percentile)
![image](https://github.com/anuraghazra/github-readme-stats/assets/141806496/a6ff7a71-d3c9-4458-9f4e-344b467713fd)
[My Read me file where it isn't](https://github.com/Rizwans-github)

**Comment by qwerty541** [comment, marked as answer]
Hey, @Rizwans-github! Weight of the commits count during rank calculation can be different depending on commits count basis, it can be yearly or all time. When using yearly basis the weight of commits count is 4x times more than when using all time basis. Here is the code: https://github.com/anuraghazra/github-readme-stats/blob/master/src/calculateRank.js#L47. On the first screenshot the commits count is calculated yearly, so the rank is higher. On the second screenshot stats card shows all time commits, so the rank is lower. All correct.

**Reply by Rizwans-github** [reply]
Thank you for the clarification, @qwerty541. I followed your suggestion and changed it to (**include_all_commits=false**) it worked perfectly. Your help is much appreciated. Thanks again!

**Reply by qwerty541** [reply]
Glad to help! Feel free to contact me if you need any further assistance.

**Reply by Rizwans-github** [reply]
I have recently started using Github, so I am trying out different things and upskilling myself. Thanks for the support, and I will definitely ask if I get stuck and can't seem to find an answer.
